modules/queue/workerpool.go

```diff
@@ -341,13 +431,55 @@ func (p *WorkerPool) FlushWithContext(ctx context.Context) error {
 
 func (p *WorkerPool) doWork(ctx context.Context) {
 	delay := time.Millisecond * 300
+
+	// Create a common timer - we will use this elsewhere
+	timer := time.NewTimer(0)
+	if !timer.Stop() {
+		select {
+		case <-timer.C:
+		default:
+		}
+	}
```

Comment:
Maybe we do not need the `select`, Either timer.C has data, or timer.C was closed.

Go document says:

```
// 	if !t.Stop() {
// 		<-t.C
// 	}
```

Reply:
No this can and will leads to blocking.

Reply:
Oh I see, the code is trying to stop the timer after it's created.

There is a `util.StopTimer(timer)`, should we use it directly?